modules/core/src/lib/draw-layers.js

```diff
@@ -247,25 +252,72 @@ function drawLayersInViewport(
   logRenderStats({renderStats, pass, redrawReason, stats});
 }
 
+function getLightSources({viewport, effects}) {
+  let lightEffect;
+  if (effects && Array.isArray(effects)) {
+    for (const i in effects) {
+      const effect = effects[i];
+      if (effect instanceof LightingEffect) {
+        lightEffect = effect;
+        break;
+      }
+    }
+  }
+
```

Comment:
I do not have much context here, if `effects` is supposed to be an `array` and has only one instance of `LightingEffect`, you can do this.

```
function getLightSources({viewport, effects = []}) {
  ...
  const lightEffect = effects.find(effect => effect instanceof LightingEffect);

```

But if `effects` is not an `array`, instead following `[Iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#Iterable_examples)` protocol, you can do the following, 

```
    for (const effect of effects) {
      if (effect instanceof LightingEffect) {
        lightEffect = effect;
        break;
      }
    }

```